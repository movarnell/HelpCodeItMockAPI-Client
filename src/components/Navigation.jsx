import React from 'react';
import { Link, useNavigate, useLocation } from 'react-router-dom';

function Navigation() {
  const navigate = useNavigate();
  const location = useLocation();

  const handleLogout = () => {
    localStorage.removeItem('token');
    navigate('/login');
  };

  const isActive = (path) => {
    return location.pathname === path ? 'border-indigo-500 text-gray-900' : 'border-transparent text-gray-500 hover:border-gray-300 hover:text-gray-700';
  };

  return (
    <nav className="bg-white shadow">
      <div className="px-4 mx-auto max-w-7xl sm:px-6 lg:px-8">
        <div className="flex justify-between h-16">
          <div className="flex">
            <div className="flex items-center flex-shrink-0">
              <Link to="/dashboard" className="text-2xl font-bold text-indigo-600">Help Mock It </Link>
            </div>
            <div className="hidden sm:ml-6 sm:flex sm:space-x-8">
              <Link
                to="/dashboard"
                className={`inline-flex items-center px-1 pt-1 text-sm font-medium border-b-2 ${isActive('/dashboard')}`}
              >
                Dashboard
              </Link>
              <Link
                to="/endpoints"
                className={`inline-flex items-center px-1 pt-1 text-sm font-medium border-b-2 ${isActive('/endpoints')}`}
              >
                Endpoints
              </Link>
              <Link
                to="/endpoints/create"
                className={`inline-flex items-center px-1 pt-1 text-sm font-medium border-b-2 ${isActive('/endpoints/create')}`}
              >
                Create Endpoint
              </Link>
            </div>
          </div>
          <div className="flex items-center">
            <button
              onClick={handleLogout}
              className="inline-flex items-center px-4 py-2 ml-3 text-sm font-medium text-white bg-indigo-600 border border-transparent rounded-md hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </nav>
  );
}

export default Navigation;