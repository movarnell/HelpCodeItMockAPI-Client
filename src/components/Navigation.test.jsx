import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';
import { BrowserRouter, useNavigate, useLocation } from 'react-router-dom';
import Navigation from './Navigation';
import renderer from 'react-test-renderer';

// Mock the react-router-dom hooks
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
  useLocation: jest.fn(),
}));

// Mock localStorage
const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn(),
};
Object.defineProperty(window, 'localStorage', { value: localStorageMock });

describe('Navigation', () => {
  const mockNavigate = jest.fn();

  beforeEach(() => {
    useNavigate.mockReturnValue(mockNavigate);
    useLocation.mockReturnValue({ pathname: '/' });
  });

  test('renders navigation links', async () => {
    await act(async () => {
      render(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      );
    });

    expect(screen.getByText('Help Mock It')).toBeInTheDocument();
    expect(screen.getByText('Dashboard')).toBeInTheDocument();
    expect(screen.getByText('Endpoints')).toBeInTheDocument();
    expect(screen.getByText('Create Endpoint')).toBeInTheDocument();
  });

  test('logout button calls handleLogout', async () => {
    await act(async () => {
      render(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      );
    });

    const logoutButton = screen.getByText('Logout');
    fireEvent.click(logoutButton);

    expect(localStorageMock.removeItem).toHaveBeenCalledWith('token');
    expect(mockNavigate).toHaveBeenCalledWith('/login');
  });

  test('active link has correct style', async () => {
    useLocation.mockReturnValue({ pathname: '/dashboard' });

    await act(async () => {
      render(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      );
    });

    const dashboardLink = screen.getByText('Dashboard');
    expect(dashboardLink).toHaveClass('border-indigo-500');
    expect(dashboardLink).toHaveClass('text-gray-900');

    const endpointsLink = screen.getByText('Endpoints');
    expect(endpointsLink).toHaveClass('border-transparent');
    expect(endpointsLink).toHaveClass('text-gray-500');
  });

  test('matches snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});