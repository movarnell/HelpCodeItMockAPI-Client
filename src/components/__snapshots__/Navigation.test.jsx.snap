// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot 1`] = `
<nav
  className="bg-white shadow"
>
  <div
    className="px-4 mx-auto max-w-7xl sm:px-6 lg:px-8"
  >
    <div
      className="flex justify-between h-16"
    >
      <div
        className="flex"
      >
        <div
          className="flex items-center flex-shrink-0"
        >
          <a
            className="text-2xl font-bold text-indigo-600"
            href="/dashboard"
            onClick={[Function]}
          >
            Help Mock It 
          </a>
        </div>
        <div
          className="hidden sm:ml-6 sm:flex sm:space-x-8"
        >
          <a
            className="inline-flex items-center px-1 pt-1 text-sm font-medium border-b-2 border-transparent text-gray-500 hover:border-gray-300 hover:text-gray-700"
            href="/dashboard"
            onClick={[Function]}
          >
            Dashboard
          </a>
          <a
            className="inline-flex items-center px-1 pt-1 text-sm font-medium border-b-2 border-transparent text-gray-500 hover:border-gray-300 hover:text-gray-700"
            href="/endpoints"
            onClick={[Function]}
          >
            Endpoints
          </a>
          <a
            className="inline-flex items-center px-1 pt-1 text-sm font-medium border-b-2 border-transparent text-gray-500 hover:border-gray-300 hover:text-gray-700"
            href="/endpoints/create"
            onClick={[Function]}
          >
            Create Endpoint
          </a>
        </div>
      </div>
      <div
        className="flex items-center"
      >
        <button
          className="inline-flex items-center px-4 py-2 ml-3 text-sm font-medium text-white bg-indigo-600 border border-transparent rounded-md hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</nav>
`;
